js/plates-model/plate.js

```diff
@@ -185,6 +185,42 @@ export default class Plate extends PlateBase {
     return count
   }
 
+  calculateContinentBuffers () {
+    const queue = []
+    const dist = {}
+    const getDist = field => {
+      const id = field.id
+      if (dist[id] !== undefined) return dist[id]
+      return Infinity
+    }
+
+    this.forEachField(field => {
+      field.isContinentBuffer = false
+      if (field.isContinent && !field.island) {
+        field.forEachNeighbour(adjField => {
+          if (adjField.isOcean && getDist(adjField) > grid.fieldDiameterInKm) {
+            dist[adjField.id] = grid.fieldDiameterInKm
+            queue.push(adjField)
+          }
+        })
+      }
+    })
+
+    while (queue.length > 0) {
+      const field = queue.shift()
+      field.isContinentBuffer = true
+      const newDist = getDist(field) + grid.fieldDiameterInKm
```

Comment:
Out of curiosity, what is the reason for using kilometers to measure distance instead of hexes?

Reply:
Note that hexes can have different dimensions. It's configurable using `divisions` param. So, the grid can be denser. And we don't want more precise model to behave differently. 

In general, I think it's always better to use some kind of world units in modeling, whenever it's possible. Or come up with some model unit system. Easier to read, understand, and it's more independent of the implementation. I can say in config that this buffer area is 750km wide, no matter what grid (or even model) we use.

Reply:
That makes a lot of sense, thanks for the explanation!